components/concept/storage/build.gradle

```diff
@@ -27,6 +27,8 @@ dependencies {
     // dependency, but it will crash at runtime.
     // Included via 'api' because this module is unusable without coroutines.
     api Dependencies.kotlin_coroutines
+
+    api project(':lib-dataprotect')
```

Comment:
Let's instead move over `KeyProvider` and `ManagedKey` into `concept-storage`. None of our concepts depend on any other modules, which is a nice pattern I'd like to keep.

These two interfaces now become part of the storage definitions, which I think makes sense here. E.g. to interact with a storage, you need to be aware of these key interfaces.

And, libdataprotect doesn't actually provide a concrete implementations for these interaces - that happens as part of our storage impl for autofill stuff. So that aligns pretty well as well - concept-storage interfaces implemented by a specific storage module.